internal/file/proto_set_provider.go

```diff
@@ -90,24 +92,30 @@ func (c *protoSetProvider) GetForFiles(workDirPath string, filePaths ...string)
 }
 
 func (c *protoSetProvider) GetMultipleForDir(workDirPath string, dirPath string) ([]*ProtoSet, error) {
-	workDirPath, err := AbsClean(workDirPath)
-	if err != nil {
+	if err := c.validateOptions(); err != nil {
 		return nil, err
 	}
-	absDirPath, err := AbsClean(dirPath)
+	workDirPath, err := AbsClean(workDirPath)
 	if err != nil {
 		return nil, err
 	}
-	configFilePath, err := c.configProvider.GetFilePathForDir(absDirPath)
+	absDirPath, err := AbsClean(dirPath)
 	if err != nil {
 		return nil, err
 	}
-	// we need everything for generation, not just the files in the given directory
-	// so we go back to the config file if it is shallower
-	// display path will be unaffected as this is based on workDirPath
-	configDirPath := absDirPath
-	if configFilePath != "" {
-		configDirPath = filepath.Dir(configFilePath)
+	configDirPath := c.configDirPath
```

Comment:
Let's remove `protoSetProvider.configDirPath` and just use the `workDirPath` instead. They should always be the same.

```
configDirPath := absDirPath
if c.configData {
  // The users has specified configuration via the command line. Set the configuration
  // directory to the current working directory.
  configDirPath = workDirPath
} else {
  configFilePath, err := c.configProvider.GetFilePathForDir(absDirPath)
  if err != nil {
    return nil, err
  }
  // We need everything for generation, not just the files in the given directory.
  // Traverse back to the config file if it is shallower.
  // The display path will be unaffected as this is based on workDirPath.
  if configFilePath != "" {
    configDirPath = filepath.Dir(configFilePath)
  }
}
...
```

This will let us drop the `validateOptions` helper, too.

Reply:
This logic is slightly wrong, you never use `absDirPath` as a value for `configDirPath` directly and the `else` is therefore not needed, but corrected it and updated.

Reply:
Also note that this now requires propagating the work dir path to one more place, but we were already propagating it.